File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if (s == null || t == null || s.length() != t.length())
            return false;

        int n = s.length();
        Map<Character, Character> mapping = new HashMap<>();
        Set<Character> mappedBefore = new HashSet<>();

        for (int i = 0; i < n; i++) {
            char c1 = s.charAt(i);
            char c2 = t.charAt(i);

            if (!mapping.containsKey(c1)) {
                if (mappedBefore.contains(c2))
                    return false;
                mapping.put(c1, c2);
                mappedBefore.add(c2);
            } else {
                if (mapping.get(c1) != c2)
                    return false;
            }
        }

        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        match={}
        for i in range(0,len(s)):
            temp=s[i]
            if temp not in match:
                if  t[i] not in match.values():
                    match[temp]=t[i]
                else:
                    return False
            
            else:
                if match[temp] != t[i]:
                    return False
        return True
                
        